sds/src/secondary_validation/lithuanian_personal_identification_number_checksum.rs

```diff
@@ -0,0 +1,72 @@
+use crate::secondary_validation::Validator;
+
+const ROUND_1_WEIGHTS: &[u32] = &[1, 2, 3, 4, 5, 6, 7, 8, 9, 1];
+const ROUND_2_WEIGHTS: &[u32] = &[3, 4, 5, 6, 7, 8, 9, 1, 2, 3];
+
+pub struct LithuanianPersonalIdentificationNumberChecksum;
+
+impl Validator for LithuanianPersonalIdentificationNumberChecksum {
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        if regex_match.chars().any(|c| !c.is_ascii_digit()) {
+            return false;
+        }
+
+        // 10 chars + 1 checksum
+        if regex_match.len() != 11 {
+            return false;
+        }
+
+        let mut checksum = run_round(regex_match, ROUND_1_WEIGHTS) % 11;
+        if checksum == 10 {
+            checksum = run_round(regex_match, ROUND_2_WEIGHTS) % 11;
+            if checksum == 10 {
+                checksum = 0;
+            }
+        }
+
+        let actual_checksum = regex_match.chars().last().unwrap().to_digit(10).unwrap();
```

Comment:
Instead of validating the length above and then using unwrap here, we could take the first 10 digits above and then use `digits.next()` to validate that the input is composed of 11 digits:

```rust
        let actual_checksum = match digits.next() {
            Some(checksum) => checksum,
            None => return false,
        };
```